pkg/services/oauth/oauth.go

```diff
@@ -1,26 +1,95 @@
 package oauth
 
 import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
 	"encoding/json"
+	"encoding/pem"
 	"errors"
 	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
 	"time"
 
 	"github.com/dgrijalva/jwt-go"
 	"github.com/google/uuid"
 	nutsCrypto "github.com/nuts-foundation/nuts-crypto/pkg"
+	"github.com/nuts-foundation/nuts-crypto/pkg/cert"
 	nutsCryptoTypes "github.com/nuts-foundation/nuts-crypto/pkg/types"
 	core "github.com/nuts-foundation/nuts-go-core"
 	nutsRegistry "github.com/nuts-foundation/nuts-registry/pkg"
+	"github.com/sirupsen/logrus"
 
 	"github.com/nuts-foundation/nuts-auth/pkg/services"
 )
 
+// ConfGenerateOAuthKeys enables key generation for JWT signing keys used in the oauth flow.
+const ConfGenerateOAuthKeys = "oAuthKeyGeneration"
+
+// ConfOAuthSigningKey is the config for where the OAuth JWT signing key is located.
+const ConfOAuthSigningKey = "oAuthSigningKey"
+
 var _ services.AccessTokenHandler = (*OAuthService)(nil)
 
 type OAuthService struct {
-	Crypto   nutsCrypto.Client
-	Registry nutsRegistry.RegistryClient
+	OAuthSigningKey   string
+	GenerateOAuthKeys bool
+	Crypto            nutsCrypto.Client
+	Registry          nutsRegistry.RegistryClient
+	signer            crypto.Signer
+}
+
+func (s *OAuthService) Configure() (err error) {
+	var bytes []byte
+	if strings.TrimSpace(s.OAuthSigningKey) == "" {
+		logrus.Warn("OAuth authorization server disabled")
+		return
+	}
+	bytes, err = ioutil.ReadFile(s.OAuthSigningKey)
+	if os.IsNotExist(err) && s.GenerateOAuthKeys {
+		bytes, err = s.generateOAuthSigningKey()
+	}
+
+	if err != nil {
+		fmt.Errorf("failed to parse OAuthSigningKey: %w", err)
+		return
+	}
+
+	s.signer, err = cert.PemToSigner(bytes)
+	return
+}
+
+func (s *OAuthService) generateOAuthSigningKey() (bytes []byte, err error) {
+	var (
```

Comment:
shouldn't this be done by `nuts-crypto` so we can switch to other backends (like vault) more easily in the future?